src/extensions/agent_modules/cloud_communication/adps2_mqtt_client/src/adps2_mqtt_client_module.c

```diff
@@ -476,84 +602,48 @@ int ADPS_MQTT_Client_Module_Deinitialize(ADUC_AGENT_MODULE_HANDLE module)
 {
     Log_Info("Deinitialize");
 
-    ADUC_CommunicationChannel_Deinitialize();
+    ADPS_MQTT_CLIENT_MODULE_STATE* moduleState = ModuleStateFromModuleHandle(module);
+    if (moduleState == NULL)
+    {
+        Log_Error("Invalid moduleState");
+        return -1;
+    }
+
+    ADUC_AGENT_MODULE_INTERFACE* commInterface = (ADUC_AGENT_MODULE_INTERFACE*)moduleState->commModule;
+    commInterface->deinitializeModule(moduleState->commModule);
 
-    FreeAzureDPS2MqttSettings(&s_moduleState.settings);
-    free(s_moduleState.operationId);
-    json_value_free(s_moduleState.registrationData);
+    FreeAzureDPS2MqttSettings(&moduleState->settings);
```

Comment:
This is code smell / bug farm because freeing fields of module state is large distance from code that is freeing the moduleState obj instance itself (when freeing module interface object).  All freeing of moduleState and its field should be in single place.  Again, concept of state store might help here.